src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/RuntimeHelpers.cs

```diff
@@ -13,22 +14,37 @@ public static partial class RuntimeHelpers
         public delegate void CleanupCode(object userData, bool exceptionThrown);
 
         /// <summary>
-        /// GetSubArray helper method for the compiler to slice an array using a range.
+        /// Slices the specified array using the specified range.
         /// </summary>
         public static T[] GetSubArray<T>(T[] array, Range range)
         {
-            Type elementType = array.GetType().GetElementType();
-            Span<T> source = array.AsSpan(range);
+            if (array == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            }
+
+            Range.OffsetAndLength offLen = range.GetOffsetAndLength(array.Length);
+            if (offLen.Length == 0)
+            {
+                return Array.Empty<T>();
```

Comment:
Err, actually, just realized we probably still only want to do this on the first path... I'll move it into the if block.